elastic/logstash/pipeline/logstash.conf

```diff
@@ -0,0 +1,138 @@
+      }
+      mutate {
+        add_field => { "read_timestamp" => "%{@timestamp}" }
+      }
+      date {
+        match => [ "[nginx][access][time]", "dd/MMM/YYYY:H:m:s Z" ]
+        remove_field => "[nginx][access][time]"
+      }
+      useragent {
+        source => "[nginx][access][agent]"
+        target => "[nginx][access][user_agent]"
+        remove_field => "[nginx][access][agent]"
+      }
+      geoip {
+        source => "[nginx][access][remote_ip]"
+        target => "[nginx][access][geoip]"
+      }
+
+      geoip {
+        source => "[nginx][access][client_ip]"
+        target => "[nginx][access][server][geoip]"
+      }
+
+
+      # decode JWT
+      if [@metadata][jwt] {
+
+
+        mutate {
+          split => { "[@metadata][jwt]" => "." }
+        }
+
+        mutate {
+          copy => { "[@metadata][jwt][1]" => "[@metadata][b64]" }
+        }
+
+        ruby {
+          init => "require 'base64'"
+          code => "event.set('[@metadata][jwt_decoded]', Base64.decode64(event.get('[@metadata][b64]')))"
+        }
+
+        json {
+          source => "[@metadata][jwt_decoded]"
+          target => "jwt_data"
+        }
+
+      }
+
+      if [nginx][access][payload] {
+        # remove \ from payload
+        mutate {
+          gsub => [
+            "[nginx][access][payload]", "[\\]", ""
+          ]
+        }
+
+        # parse json to data field
+        json {
+          source => "[nginx][access][payload]"
+          target => "[data]"
+        }
+      }
+
+
+      # Get the type of operation of interest
+
+      mutate {
+        add_field => { "[@metadata][action]" => "%{[nginx][access][method]} %{[nginx][access][url]}" }
+      }
+
+      # Get the operation
+      translate {
+        field => "[@metadata][action]"
+        destination => "operation"
+        dictionary_path => "/usr/share/logstash/dictionaries/action.yml"
+        regex => true
+        fallback => "other"
+      }
+
+      # Extract mine guid based on operation
+      if [operation] != "other" {
+        grok {
+          match => { "[nginx][access][url]" => [".*/api/mines/%{UUID:[mine][id]}", ".*/568/api/documents/mines/%{UUID:[mine][id]}"] }
+        }
+      }
+
+    }
+    else if [fileset][name] == "error" {
+      grok {
+        match => { "message" => ["%{DATA:[nginx][error][time]} \[%{DATA:[nginx][error][level]}\] %{NUMBER:[nginx][error][pid]}#%{NUMBER:[nginx][error][tid]}: (\*%{NUMBER:[nginx][error][connection_id]} )?%{GREEDYDATA:[nginx][error][message]}"] }
+        remove_field => "message"
+      }
+      mutate {
+        rename => { "@timestamp" => "read_timestamp" }
+      }
+      date {
+        match => [ "[nginx][error][time]", "YYYY/MM/dd H:m:s" ]
+        remove_field => "[nginx][error][time]"
+      }
+    }
+  }
+}
+
+
+output {
+  
+  # stdout {}
+
+  elasticsearch {
+    hosts => "${ES_HOST}"
+    user => "${ES_USER}"
+    password => "${ES_PASSWORD}"
+    index => "mds-nginx-%{+YYYY.MM.dd}"
```

Comment:
Can you make this environment specific? Something like mds-nginx-dev-292-timestamp. We can provide that prefix in an environment variable.
Just so we can keep indexes separate for dev PRs, test and prod.
Although im not sure, that information may already be in the nginx logs